foundry/foundry.py

```diff
@@ -122,10 +126,30 @@ def load(self, name, download=True, globus=True, verbose=False, metadata=None, *
         else:
             res = metadata
 
-        # TODO: if object empty, handle
-        res = res[0]
-        res["dataset"] = res["projects"][self.config.metadata_key]
-        res["dataset"]["type"] = res["dataset"]["data_type"]
+        # we figured from our tests that at each of these three lines there could be an error
+        # if res is an empty list. (we also see that res will always be an empty list,
+        # never None or not a list)
+        try:
+            res = res[0] # if search returns multiple results, this automatically uses first result
+        except IndexError as e:
+            raise Exception("load: No metadata found for given dataset") from e
+        except BaseException as e:
+            raise Exception("load: An unknown error occurred") from e
```

Comment:
you don't need to except the `BaseException` each time, since that'll automatically be raised and won't be caught by the `IndexError`. And we don't have any more information to give the user for this unknown exception (they will already know it happened in the `load()` function because it'll be part of the stack trace)